File: 🟡 Medium/2657 - Find the Prefix Common Array of Two Arrays/_2657FindthePrefixCommonArrayofTwoArrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n = A.length;
        int freq[] = new int[n + 1];
        int res[] = new int[n];

        int count = 0;

        for (int i = 0; i < n; i++) {
            freq[A[i]]++;
            if (freq[A[i]] == 2) {
                count++;
            }

            freq[B[i]]++;
            if (freq[B[i]] == 2) {
                count++;
            }

            res[i] = count;
        }

        return res;
    }
}

File: 🟡 Medium/2657 - Find the Prefix Common Array of Two Arrays/_2657FindthePrefixCommonArrayofTwoArrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        freq = [0] * (n + 1)
        res = [0] * n

        count = 0

        for i in range(0, n):
            freq[A[i]] += 1
            if freq[A[i]] == 2:
                count += 1

            freq[B[i]] += 1
            if freq[B[i]] == 2:
                count += 1

            res[i] = count

        return res